Add timeout option for nxapi

dgjustice opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

nxos.py
##### ANSIBLE VERSION

```
ansible 2.2.0 (timeout bebb6b9fb4) last updated 2016/10/17 14:31:47 (GMT -500)
  lib/ansible/modules/core: (detached HEAD 488f082761) last updated 2016/10/17 14:20:51 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD 24da3602c6) last updated 2016/10/17 14:21:14 (GMT -500)
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

This PR allows you to configure a custom timeout value for transport using nx-api.  PR 14973 added timeout to ios, and those changes are carried into the cli transport for nxos (whether or not by design :) ).  I added code to set timeout in the url module as well.

```

---
- hosts: all
  gather_facts: no
  connection: local
  tasks:
  - name: return show commands output
    nxos_command:
      host: "{{ ansible_host }}"
      username: user
      password: password
      transport: nxapi
      commands:
        - show version
      timeout: 30
    register: output

  - debug: var=output
```

```
modified:   lib/ansible/module_utils/nxos.py
- added configurable timeout to module paramaters
modified:   lib/ansible/utils/module_docs_fragments/nxos.py
- added documentation for timeout
```

Commit:
Changes to be committed:
	modified:   lib/ansible/module_utils/nxos.py
    - added configurable timeout to module paramaters
	modified:   lib/ansible/utils/module_docs_fragments/nxos.py
    - added documentation for timeout

Commit:
Changes to be committed:
    modified:   ansible/module_utils/nxos.py
    - added timeout option for nxapi transport and added documentation
    - option works with CLI or NXAPI transport

gundalow commented on lib/ansible/utils/module_docs_fragments/nxos.py:
```diff
@@ -73,6 +73,12 @@ class ModuleDocFragment(object):
     choices: ['yes', 'no']
+  timeout:
+    description:
+      - Specifies idle timeout for the connection.  NX-API can be slow to 
```
Can you please add the units, e.g.
"Specifies idle timeout for the connection, in seconds."

gundalow commented on lib/ansible/utils/module_docs_fragments/nxos.py:
```diff
@@ -73,6 +73,12 @@ class ModuleDocFragment(object):
+      - Specifies idle timeout for the connection.  NX-API can be slow to 
+        return on long-running commands (sh mac, sh bgp, etc).
+    required: false
+    default: 10
```
Will need to add a `version_added: 2.3` in here.

Or 2.2. if @privateip is happy to accept and cherry pick this change into `stable-2.2`

gundalow commented:
Assigning to @privateip to decide if we want this simple (though useful) feature into 2.2 (i.e. cherry pick) or into 2.3 (devel)

Commit:
Changes to be committed:
	modified:   lib/ansible/utils/module_docs_fragments/nxos.py
    - Changed per comments in PR 18074

dgjustice commented:
I added comments to the documentation per your requests.  Specified timeout in seconds and version_added 2.3.

Commit:
Changes to be committed:
	modified:   lib/ansible/module_utils/nxos.py
    - added try/except block to test for timeout

Commit:
Changes to be committed:
modified:   lib/ansible/module_utils/nxos.py
 - tweaked timeout

gundalow commented:
Merged, thanks
